lib/cchain/provider/attestations.go

```diff
@@ -0,0 +1,350 @@
+package provider
+
+import (
+	"context"
+
+	atypes "github.com/omni-network/omni/halo/attest/types"
+	"github.com/omni-network/omni/lib/errors"
+	"github.com/omni-network/omni/lib/log"
+	"github.com/omni-network/omni/lib/tracer"
+	"github.com/omni-network/omni/lib/umath"
+	"github.com/omni-network/omni/lib/xchain"
+
+	"github.com/cosmos/cosmos-sdk/client/grpc/cmtservice"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+type attestations struct {
```

Comment:
not a fan of this OO style of coding, why can't we use simple functions?

Reply:
also the type name doesn't really make sense?

Reply:
I guess is a bit of a preference, I just thought there was a lot of passing of clients around the functions, I can go back to functions and test those directly